### PR TITLE
adding events and fixing potential thread issue

### PR DESCRIPTION
**Pull request by imhotep**

### CODE REVIEW

**Review by yuliyantsvetkov — APPROVED** [review]
LGTM. Thanks for your integration!

File: custom_components/unifi_access/event.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+class AccessEventEntity(EventEntity):
+    """Authorized User Event Entity."""
+
+    _attr_event_types = [ACCESS_ENTRY_EVENT, ACCESS_EXIT_EVENT]
+
+    def __init__(self, hass: HomeAssistant, door) -> None:
+        """Initialize Unifi Access Door Lock."""
+        self.hass = hass
+        self.id = door.id
```
Comment (blablacio) [inline custom_components/unifi_access/event.py:53]: Do we need this here? I see `self.door.id` used in other methods...

File: custom_components/unifi_access/hub.py [hunk 2]
```diff
@@ -260,14 +272,64 @@ def on_message(self, ws: websocket.WebSocketApp, message):
...
                             "Doorbell press stopped on %s Request ID %s",
                             existing_door.name,
                             doorbell_request_id,
                         )
-
+                case "access.logs.add":
+                    door = next(
+                        (
+                            target
+                            for target in update["data"]["_source"]["target"]
```
Comment (blablacio) [inline custom_components/unifi_access/hub.py:290]: Are we sure all the keys are in place here?

File: custom_components/unifi_access/hub.py [hunk 3]
```diff
@@ -260,14 +272,64 @@ def on_message(self, ws: websocket.WebSocketApp, message):
...
+                            if target["type"] == "door"
+                        ),
+                        None,
+                    )
+                    if door is not None:
+                        door_id = door["id"]
+                        _LOGGER.info("Access log added via websocket %s", door_id)
+                        if door_id in self.doors:
+                            existing_door = self.doors[door_id]
+                            actor = update["data"]["_source"]["actor"]["display_name"]
```
Comment (blablacio) [inline custom_components/unifi_access/hub.py:300]: Same as above.

File: custom_components/unifi_access/hub.py [hunk 4]
```diff
@@ -260,14 +272,64 @@ def on_message(self, ws: websocket.WebSocketApp, message):
...
+                            device_config = next(
+                                (
+                                    target
+                                    for target in update["data"]["_source"]["target"]
```
Comment (blablacio) [inline custom_components/unifi_access/hub.py:304]: Same as above.

**Review by blablacio — COMMENTED** [review]
Looks good apart from the comments.

File: custom_components/unifi_access/hub.py (hunk 2 above)
Comment (imhotep) [reply custom_components/unifi_access/hub.py:290]: Yes. It's part of the response schema and the keys are always there. I couldn't find a clean way to conditionally access properties in Python (short of using `get` followed by None check).

**Comment by imhotep** [comment]
@yuliyantsvetkov @blablacio thanks for reviewing :)

**Pull request merged**